### ISSUE TITLE
Draft changes to text select tab

### ISSUE THREAD

**Issue by davidskalinder** [davidskalinder/mpeds-coder#8]
1. Actors  [[who is protesting]]
   1. Names or descriptions of actors
   2. Text relevant to number of participants or event size [[both numbers and phrases like "hundreds" or "filled council chambers" or "busloads"
   3. Named organizations
2. Issue [[reason for or purpose of event]]
3. Target [[who is being asked to do something]] 
4. Bystanders [[do we need this? It is rarely used]]
5. Actions by protesters
   1. violence to persons or property
   2. other actions
6. Police actions (includes arrests or statement)
7. Location
   1. City/Place name
   2. Specific place in city (e.g. council chambers)

Omit square brackets if necessary for space.

**Comment by davidskalinder** [comment]
Single-level items will be easy to implement in text-select.yaml.  Not sure yet how to handle multi-level items.

**Comment by davidskalinder** [comment]
NB: presets tab has/had functionality for radios to open new sets of checkboxes at https://github.com/davidskalinder/mpeds-coder/blob/570b5bc098ef9e1ec81b4f04f1a1237033b01643/static/ec.js#L43)

Maybe this could be ported to this issue to provide at least two levels of controls.

**Comment by davidskalinder** [comment]
@olderwoman, @matthewsmorgan has reminded me that we're getting to the point where we probably need to make a call on how we want this list to be in the production version.

Also, just to clarify here: if we agree on a one-level list for this (i.e., a list of categories with no subcategories), then the implementation will only take a few seconds.  If we need a two-level list then I'll have to check whether this is an easy fix or a harder one.

**Comment by davidskalinder** [comment]
Also, to better-record something I put in an email about this a long time ago:

---

We've discussed the idea of simply asking pass 1 coders to highlight only enough text to create a sort of abbreviated account of the event for pass 2 coders to read, but dismissed it since it would create an incentive to simply highlight the whole article. However, I wonder whether it's really necessary to ask them to parse all 12 of the current categories (target, police, actor, politicians, orgs, etc. -- since the goal is, I think, indeed merely to give pass 2 people enough text to make their decisions)? Perhaps we could simply ask/require pass 1 coders to highlight text that answers something like [Kipling's questions](http://www.kiplingsociety.co.uk/poems_serving.htm): who was involved, what did everybody want, what did everybody do, what were the immediate consequences, etc.?  (Perhaps also requiring that they check a "not provided" box for each question not answered in the article?)

I'd also note that if we simplify the highlighting like this and make the other cuts suggested in the doc, then we're getting toward a pretty tidy structure: a description and a couple other fields about the article, 3 fields to justify whether a thing is a codeable event or not, a description and 5 fields about each codeable event, and then enough text capture to recreate a minimal account of the event.

---

Happy to be overruled on that, of course, but that's my $0.02.

**Comment by olderwoman** [comment]
Yes, especially as we are finding that my prior pass one coders missed a lot of info. We need to figure out what should and should not be marked in pass 1. However, the pass 2 interface is not good for marking text and having the text marked (properly) is an asset. So important to figure out how to balance things.

**Comment by davidskalinder** [comment]
1. Names or characteristics of people involved in the protest
1. Named organizations
1. Text relevant to number of participants or event size
1. Issue (reason for or purpose of event)
1. Target (who is being asked to do something)
1. Actions by protesters (violent or otherwise)
1. Police actions (includes arrests or statement)
1. Location

**Comment by davidskalinder** [comment]
List is good enough for now -- put in testing instance, then bug @olderwoman (and @matthewsmorgan) daily until they've tested it.

**Comment by davidskalinder** [comment]
@olderwoman / @matthewsmorgan, this should now be updated in the testing deployment as well.  Again, looks okay to me but maybe a little bit too wordy.  (Perhaps if we get issue #24 working at some point some of this wordiness can be moved into tooltips?)

**Comment by davidskalinder** [comment]
Ah, another (probably low-ish priority) thing for me to do on this: the items are sorting strangely (specifically, alphabetically by the name of the invisible variable I've given them).  I should load the YAML using the new ordered_load function so we can customize it.

**Comment by davidskalinder** [comment]
@matthewsmorgan sez:

>     1. Text Select
>        I'd suggest shortening three of these items:
>        Actions by Protesters
>        Size/Number of Participants
>        Target

Sounds good to me.  @olderwoman?

**Comment by olderwoman** [comment]
I was happy with the text selection options but labels are too long. Plus alphabetized rather than in a good order. "Text relevant . . . size" can be "Size relevant text". Issues I think can safely just be issue. Actions by protesters just that.

**Comment by davidskalinder** [comment]
Okay, so combining @matthewsmorgan's and @olderwoman's suggestions, I think we get to:

1. Actors
1. Bystanders
1. Named organizations
1. Size-relevant text
1. Issue
1. Target
1. Actions by protesters
1. Anything about police
1. Location
1. Time/duration

The alphebetization is something I think I know how to to but it'll be trickier than changing labels so I'm putting it off while I do more urgent stuff if that sounds sensible to you?

**Comment by davidskalinder** [comment]
New issue for the sort order: #30

**Comment by matthewsmorgan** [comment]
It does not look like on my end that the proposed changes, shortening the text (see above), were made. I'd suggest shortening to Actions by Protesters, Issue, Size-Relevant Text, and Target. The other text selects seems fine to me.

**Comment by davidskalinder** [comment]
> It does not look like on my end that the proposed changes, shortening the text (see above), were made. I'd suggest shortening to Actions by Protesters, Issue, Size-Relevant Text, and Target. The other text selects seems fine to me.

Hmm, yeah looks like you're right.  I wonder why I forgot to do this? . . .  Anyway, it should be fixed now in the testing deployment.

I also took the liberty of changing the police item to "Police actions during protest" to address the concern that arose during training on Friday.  Please let me know if we want a different wording!

One other word on this: the text for this tab (and the yes/no tab) is set in a config file not under version control.  What means is that it's possible in theory that this might go wrong again if the wrong config file ends up in the production folder somehow.  I'll try to make sure only the correct ones are there, but please yell at me if you see that it's wrong again!  (The fix, once we notice it, is very easy.)

**Comment by davidskalinder** [comment]
Hey look at that I followed my own advice here even though I didn't in #50.  Anyway, looks good in production now so I'm closing this.
